Library/Homebrew/cmd/upgrade.rb

```diff
@@ -41,7 +42,10 @@ def upgrade
       puts pinned.map { |f| "#{f.full_name} #{f.pkg_version}" } * ", "
     end
 
-    outdated.each { |f| upgrade_formula(f) }
+    outdated.each do |f|
+      upgrade_formula(f)
+      cleanup_formula(f) if ARGV.include? "--cleanup"
```

Comment:
We might want to cache this value, e.g.:

```rb
cleanup = ARGV.include? "--cleanup"
outdated.each do |f|
  upgrade_formula(f)
  cleanup_formula(f) if cleanup
end
```

That’s probably unnecessary optimization thought.